src/components/common/mediaPicker/results/SourceSearchResultsContainer.js

```diff
@@ -180,36 +175,42 @@ class SourceSearchResultsContainer extends React.Component {
       </Col>
     );
 
+    let resultContent;
+    let getMoreResultsContent;
     if (fetchStatus === FETCH_ONGOING) {
       resultContent = <LoadingSpinner />;
     } else if (sourceResults && (sourceResults.list && (sourceResults.list.length > 0 || (sourceResults.args && sourceResults.args.media_keyword) || (sourceResults.args && sourceResults.args.tags)))) {
       const previouslySearchedTags = {};
       const tagNames = Object.keys(selectedMediaQueryTags)
         .filter(t => metadataQueryFields.has(t) > 0 && Array.isArray(selectedMediaQueryTags[t]) && selectedMediaQueryTags[t].length > 0)
         .map((i) => {
-          const obj = selectedMediaQueryTags[i];
+          const tag = selectedMediaQueryTags[i];
           if (sourceResults.args.tags) { // correlate searched tag ids with objects so we can display the labels
-            if (!previouslySearchedTags[i]) previouslySearchedTags[i] = [];
-            previouslySearchedTags[i] = obj.map(t => ( // if in tags, it is selected, so reflect this
-              sourceResults.args.tags.indexOf(t.tags_id) > -1 ? ({ ...t, selected: true }) : ''
-            ));
+            if (!previouslySearchedTags[i]) {
+              previouslySearchedTags[i] = [];
+            }
+            previouslySearchedTags[i] = tag
+              .filter(t => sourceResults.args.tags.indexOf(t.tags_id) > -1)
```

Comment:
Filtering the results was key to fixing the `null`. Previously, this was inserting an empty string, which was converted into `null` in `explorerUtils::stringifyTags()`.

Reply:
nice and thanks! it's hairy in there

Reply:
So many things to consider!